### PR TITLE
Don't instantiate Imagick in constructor

### LINKED ISSUE #138: Problem with large file.
Server is down with 503 error when loading big pdf(8Mb, 40 pages).
It's because you puting file in construct method Imagick.

`$this->imagick = new Imagick($pdfFile);`

But why whole file if i need only one page. In method getImageData there is instruction that page is need use for generate preview.

`$this->imagick->readImage(sprintf('%s[%s]', $this->pdfFile, $this->page - 1));`

If you need read all file don't indicate index of page. Like this:

`$this->imagick->readImage($this->pdfFile);`

After then i deleted file from constructor Imagick

`$this->imagick = new Imagick();`

And all works fine. If you want pick another page there is method setPage

### PR DESCRIPTION
**Pull request by robertogallea**
This PR fixes #138 by adding a new method for fetching page count. It directly inspects pdf content.

Imagick instantiation is done without passing pdf file in constructor.

### CODE REVIEW

File: src/Pdf.php [hunk 1]
```diff
@@ -215,4 +215,24 @@ protected function determineOutputFormat(string $pathToImage): string
 
         return $outputFormat;
     }
+
+    private function fetchNumberPages(string $pdfFile)
+    {
+        $stream = fopen($pdfFile, 'r');
+        $content = fread($stream, filesize($pdfFile));
```
Comment (freekmurze) [inline src/Pdf.php]: I don't understand how you can determine the number of pages using `file size`

Comment (truechernyshov) [reply src/Pdf.php]: Not all PDF files contain "Count". Accordingly this method don't work correctly on some files.

Comment (freekmurze) [reply src/Pdf.php]: Right...

**Comment by freekmurze** [comment]
Closing this as it would work for all PDFs

**Pull request closed without merging**